accord-core/src/main/java/accord/local/CommandStore.java

```diff
@@ -470,14 +470,14 @@ final Supplier<EpochReady> bootstrapper(Node node, Ranges newRanges, long epoch)
      */
     protected Supplier<EpochReady> sync(Node node, Ranges ranges, long epoch, boolean isLoad)
     {
-        return () -> syncInternal(node, ranges, epoch, isLoad);
+        return () -> syncInternal(node, ranges, epoch, DONE, isLoad);
     }
 
-    protected EpochReady syncInternal(Node node, Ranges ranges, long epoch, boolean isLoad)
+    protected EpochReady syncInternal(Node node, Ranges ranges, long epoch, AsyncResult<Void> metadata, boolean isLoad)
```

Comment:
is this safe?

Reply:
Right now we run the sync on the command store and block, which will 100% of the time timeout causing the syncs to keep failing. The calling path blocks then returns `DONE` to the caller, so by sequencing the work then overriding the metadata result we should yield the same result.  So if there is a caller that expects the blocking, then that caller is unsafe but could not work right now.

Reply:
I don’t follow all of that, but I’m not sure that it’s better to make something appear to pass if it’s still broken.

I am not up to speed on all of this, but my intuition is that my proposed fix to AbstractConfigurationService would resolve all of these problems, so wondering if that has been tried yet?

anyway, syncInternal is defunct now so these changes aren’t obviously needed. We should probably revert them and take another look after rebase.

Reply:
> I am not up to speed on all of this, but my intuition is that my proposed fix to AbstractConfigurationService would resolve all of these problems, so wondering if that has been tried yet?

The problem here is that C* submits a task to the CommandStore that it currently is running in, then blocks on it for 5 minutes... since its a single threaded executor this times out 100% of the time... this doesn't have anything to do with `AbstractConfigurationService` locks

Reply:
`syncInternal` is not invoked by the command store thread, or shouldn't be. We should add a check to validate that it is not, although I think we no longer depend on this. Either way, the traces I saw showed a calls via `AccordConfigurationService` callbacks, with a callback blocking while holding the ACS lock so that another thread could not make progress (while waiting on ACS).